contrib/babelfishpg_tsql/src/databasepropertyex.c

```diff
@@ -210,6 +210,11 @@ databasepropertyex(PG_FUNCTION_ARGS)
 	{
 		const char *ret = "READ_WRITE";
 
+		if (strcmp(GetConfigOption("transaction_read_only", true, false), "on") == 0) 
+		{
+			ret = "READ_ONLY";
+		}
```

Comment:
Should we also check for dialect, since from postgres endpoint I could be on the writer node but still in a read only transaction.